text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,278 @@
+cross-language documentation experience to help developers integrate those
+constructs into their application.
+
+Customers can also deploy a private instance of CDK Construct Hub to facilitate
+sharing infrastructure abstractions published to a private package registry
+within the organization.
+
+### Where can I find the CDK Construct Hub?
+
+The public instance of CDK Construct Hub is at: [awscdk.io](https://awscdk.io/).
+
+### How do I publish a package to CDK Construct Hub?
+
+Packages published to the public npm registry (npmjs.com) are automatically
+indexed by the public instance of CDK Construct Hub as long as they meet the
+following prerequisites:
+
+1. Cross-language support has been configured correctly using [jsii]
+1. The `cdk` keyword is referenced (in the `keywords` key of `package.json`)
+
+[jsii]: https://aws.github.io/jsii
+
+### Can I opt packages out of the public instance of CDK Construct Hub?
+
+There is currently no self-service way to un-list a package from the CDK
+Construct Hub. If you would like your package to be removed from the public
+instance of CDK Construct Hub, please file an issue in GitHub. You will be
+required to prove ownership of the package(s) you are requesting un-listing for.
+
+### Can I trust all constructs listed on CDK Construct Hub?
+
+CDK Construct Hub does not perform a comprehensive audit of packages it indexes.
+You should not trust packages found via the CDK Construct Hub any differently
+from packages found via other package search tools or registries, such as
+npmjs.com, Maven Central, PyPI, NuGet, etc.
+
+## Specification
+
+### Landing Page
+
+> [Example rendering](#rendering-landing-page)
+
+The landing page for the CDK Construct Hub includes a search bar that allows
+developers to search for constructs that solve their use-case using relevant
+keywords, and optionally filtering on a specific programming language and CDK
+flavor (AWS CDK / CDK8s / CDK-Tf).
+
+It may also feature a short-list of packages, selected based on popularity or
+latest release date.
+
+### Search Results Page
+
+When a search is performed using the search bar in the landing page (or in the
+header of all other pages), a list of relevant results is displayed as tiles,
+similar to the short-list of packages shown on the landing page.
+
+### Package Detail Page
+
+> [Example rendering](#rendering-package-detail-page)
+
+Clicking on a given package's tile in the landing page or search result page
+brings the user to a package detail page that represents the following
+information:
+
+- Package name, tags, and description
+- Overview of publishing information (author, version, license, etc...)
+  - Where possible, links to the source repository, issue tracker, etc...
+- Install instructions
+- Available languages
+- Package's documentation
+  - Rendered content of the package's `README.md`
+  - API reference documentation, in all supported languages
```

Comment:
Language selection should not only impact API reference documentation but also the README, installation instructions, etc. So we wanted this to be a "global" language switcher.